### PR TITLE
[Merged by Bors] - feat: quick version of mono tactic

### PR DESCRIPTION
**Pull request by hrmacbeth**
This is an extremely partial port of the `mono*` tactic from Lean 3, implemented as a macro on top of `solve_by_elim`.  The original `mono` had many configuration options and no documentation, so quite a bit is missing (and almost all the Lean 3 tests fail).  Nonetheless I think it's worth merging this, because
- it will get rid of errors in mathport output which come from lemmas being tagged with a nonexistent attribute `@[mono]`
- in most mathlib3 uses of mono, only the basic version was used, not the various configuration options; thus I would guess that this version of `mono` will succeed fairly often in the port even though it fails nearly all the tests

---

Note: the `solve_by_elim` behavior is closer to `mono*` than `mono`, so both `mono` and `mono*` are accepted here and, for now, mean the same thing.

The full port of `mono` will eventually work in every place that mathlib3 had `mono`, so as long as no `mono*` is changed to `mono` between now and the full port of `mono`, this conflation of `mono*` and `mono` shouldn't introduce many wrinkles. Any wrinkles it does introduce will be able to be easily corrected by changing `mono` to `mono*`.

`@[mono]`, `@[mono left]`, `@[mono right]`, and `@[mono both]` all just mean `@[mono]` for now.

The full port of mono will be PR'd as #2323.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by fpvandoorn** [comment]
This looks good to me, but I'm a bit worried about the syntax being different (and duplicating) the syntax in `Mathport/Syntax`. Is it possible that this trips up mathport? Maybe it's safer to implement exactly the syntax from `Mathport/Syntax` (removing it there), but ignoring all the other options.

**Comment by digama0** [comment]
There should not be any duplicate syntaxes. When partially implementing a tactic, you should copy the syntax line from `Mathport.Syntax` into the file, and then `throwUnsupportedSyntax` or ignore any flags you aren't planning on supporting yet.

File: Mathlib/Tactic/Monotonicity/Attr.lean [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/-
+Copyright (c) 2023 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import Mathlib.Tactic.LabelAttr
+
+/-! # The @[mono] attribute -/
+
+namespace Mathlib.Tactic.Monoticity
```
Comment (gebner) [inline Mathlib/Tactic/Monotonicity/Attr.lean:10]:
```suggestion
namespace Mathlib.Tactic.Monotonicity
```

File: Mathlib/Tactic/Monotonicity/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+count these as separate steps, throwing off the count in the desired configuration
+`maxDepth := 1`.  So instead we just implement a version of `mono` in which monotonicity rules,
+local hypotheses and `rfl` are all applied repeatedly until nothing more is applicable.  The syntax
+for this in Lean 3 was `mono*`. Both `mono` and `mono*` implement this behavior for now.
+-/
+
+open Lean Elab Tactic Parser Tactic
+open Mathlib Tactic SolveByElim
+
+namespace Mathlib.Tactic.Monoticity
```
Comment (gebner) [inline Mathlib/Tactic/Monotonicity/Basic.lean:30]:
```suggestion
namespace Mathlib.Tactic.Monotonicity
```

**Review by gebner — COMMENTED** [review]
bors d+

**Comment by gebner** [comment]
bors d=thorimur

**Pull request closed without merging**